sdk/python/feast/permissions/security_manager.py

```diff
@@ -101,14 +101,19 @@ def assert_permissions(
         FeastObject: The original `resource`, if permitted.
 
     Raises:
-        PermissionError: If the current user is not authorized to eecute the requested actions on the given resources (and `filter_only` is `False`).
+        PermissionError: If the current user is not authorized to execute the requested actions on the given resources (and `filter_only` is `False`).
     """
     sm = get_security_manager()
     if sm is None:
         return resource
-    return sm.assert_permissions(
+    resources = sm.assert_permissions(
         resources=[resource], actions=actions, filter_only=False
-    )[0]
+    )
+
+    if len(resources) == 0:
```

Comment:
`sm.assert_permissions` already raises the `PermissionError` if `filter_only` is `False` and there is no permission issue. IMO, if the list is empty at this point, it's because there are no resources.

Reply:
We need to talk about this in the sync

Reply:
agree to raise error, if there are no permissions (not if `len(resources)` is 0)

Reply:
Is not relevant anymore, there is no scenario, if there are no permissions at all it will return the resource and if there are no matching permission we throw a PermissionError and no need for this check anymore